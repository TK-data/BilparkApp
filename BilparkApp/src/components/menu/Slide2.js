import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet } from 'react-native';
import { View, Button, Text } from 'native-base';

const styles = StyleSheet.create({
  buzzWord: {
    color: 'white',
    fontSize: 30,
    fontWeight: 'bold',
    marginBottom: '5%',
  },
  carHeader: {
    color: 'white',
    fontSize: 22,
    fontWeight: 'bold',
  },
  text: {
    color: 'white',
    fontSize: 16,
  },
  finnBil: {
    color: 'white',
    fontSize: 18,
    marginBottom: '8%',
  },
  slide1: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(000, 039, 118)',
  },
  button: {
    alignSelf: 'center',
  },
});


class Slide2 extends Component {

  render() {

    let company = false;
    if (this.props.company) {
      company = JSON.parse(this.props.company);
    }
    const localCompany = this.props.localCompany;

    return (
      <View style={styles.slide1}>
        <Text style={styles.buzzWord}>Du tilh√∏rer</Text>
        <Text style={styles.text}>
          { company ? company.CompanyName : localCompany.CompanyName }
        </Text>
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.auth.user,
    company: state.auth.company,
    localCompany: state.registerCompany.company,
  };
};

const mapDispatchToProps = dispatch => ({
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Slide2);
