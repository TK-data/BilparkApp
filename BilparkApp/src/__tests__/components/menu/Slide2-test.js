import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Slide2 from '../../../components/menu/Slide2';

const initialState = {
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: '{"UserID":111,"Email":"awkdnawpidn@aiwfaf.no","CompanyID":7,"CarID":82,"Address":"Oiabfoaiwbf","Fname":"Obafiobafiob","Lname":"Oabfoaib","FuelDay":0,"FuelTime":"12-00","FuelNotification":false,"createdAt":"2018-05-20T11:30:10.000Z","updatedAt":"2018-05-20T11:30:21.000Z"}',
  },
  registerCompany: {
    companies: [],
    company: '',
    hasErrored: false,
    isLoading: false,
    selectedCompany: '',
  },

};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing Slide2', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <Slide2 />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
