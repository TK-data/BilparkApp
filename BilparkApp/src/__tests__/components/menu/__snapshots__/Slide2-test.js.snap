// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Slide2 Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slide2
    company={undefined}
    localCompany=""
    user="{\\"UserID\\":111,\\"Email\\":\\"awkdnawpidn@aiwfaf.no\\",\\"CompanyID\\":7,\\"CarID\\":82,\\"Address\\":\\"Oiabfoaiwbf\\",\\"Fname\\":\\"Obafiobafiob\\",\\"Lname\\":\\"Oabfoaib\\",\\"FuelDay\\":0,\\"FuelTime\\":\\"12-00\\",\\"FuelNotification\\":false,\\"createdAt\\":\\"2018-05-20T11:30:10.000Z\\",\\"updatedAt\\":\\"2018-05-20T11:30:21.000Z\\"}"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Text)
          style={
            Object {
              "color": "white",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": "5%",
            }
          }
          virtual={undefined}
        >
          Du tilhører
        </Styled(Text)>,
        <Styled(Text)
          style={
            Object {
              "color": "white",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(000, 039, 118)",
        "flex": 1,
        "justifyContent": "center",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Du tilhører",
          "style": Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": "5%",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "Du tilhører",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "color": "white",
            "fontSize": 16,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 30,
                "fontWeight": "bold",
                "marginBottom": "5%",
              }
            }
            virtual={undefined}
          >
            Du tilhører
          </Styled(Text)>,
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "rgb(000, 039, 118)",
          "flex": 1,
          "justifyContent": "center",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Du tilhører",
            "style": Object {
              "color": "white",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": "5%",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Du tilhører",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "color": "white",
              "fontSize": 16,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
